### DISCUSSION TITLE
Nextcloud AIO + nginx proxy manager for a noob

### DISCUSSION THREAD

**Discussion by lukasig** [nextcloud/all-in-one#681] [Questions] [answered]
Hi

I am a bit of a noob, but I have successfully used [nextcloudPI](https://github.com/nextcloud/nextcloudpi) for years running on a raspberry pi.
Now I upgraded my hardware to a x86 processor, I have installed ubuntu server 22.04, and I installed [nextcloud AIO (all-in-one) ](https://github.com/nextcloud/all-in-one) through docker. I want to have a few other web servers running on the same machine and that is why i also want to use [nginx proxy manager](https://nginxproxymanager.com/).

I read [here](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md) of the changes needed to run behind a proxy so I have used this command to run nextcloud AIO in docker:
`sudo docker run -d --name nextcloud-aio-mastercontainer --restart always -p 8080:8080 -e APACHE_PORT=11000 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest`

This installed correctly and I can access the AIO web interface. However I do not know how to implement the changes also needed on the nginx side, since I use the nginx proxy manager through a web interface.

Can anyone help?

thanks

PS - I had success with installing classic nextcloud (not AIO) + nginx proxy manager, and also just installing nextcloud AIO. But I am trying to install nextcloud AIO + nginx proxy manager.

**Comment by szaimen** [comment]
Hello, I am currently working on adding an example for nginx-proxy-manager to the documentation. You can see the progress here: https://github.com/nextcloud/all-in-one/pull/682

**Comment by szaimen** [comment, marked as answer]
Hello, please read through and follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md. We also have an example for nginx-proxy manager now :)

**Reply by lu-kno** [reply]
Hi, I am trying to set up my nginx-proxy-manager with nextcloud aio like in the guide, but i cant seem to be able to connect to nc after setting up nginx and starting the nextcloud containers from the mastercontainer. 

both nginx and nc-aio are created from the same docker-compose file. Does this change anything from how i need to follow the guide? I connect nginx to my other services using the name of their container as hostname, like "homeassistant:8123". this way i dont have to deal mapping ports and its all on the same network. I would love to be able to do this with nc-aio, but i dont know how, if its even possible.

**Reply by Zoey2936** [reply]
it is not possible, you need to use localhost and network mode host

**Reply by Blackwidow-sudo** [reply]
Could you explain why it is not possible?
Bcs i would also like to use a dedicated docker network for my services behind NPM. And i see no reason why it should not be possible.

**Reply by 4lexRed** [reply]
It is totally possible.
You just habe to be aware that if you use the docker container names for forwarding, the container changes during the setup.
[See here](https://github.com/nextcloud/all-in-one/discussions/4240) for a detailed explanation of what happens during the setup and how you have to configure NPM during the setup.

**Reply by bethnard** [reply]
Sorry to revive this discussion, but I was facing issues with the authentication of apps (android, desktop, etc.) for my Nextcloud AIO instance and found out to be an issue of my NPM Access List configuration. The current link on github shows how to configure it for a public access list. In case someone is using a custom acess list, the person needs to keep in mind following things:
1) add rule to accept 172.16.0.0/12 (common docker network ips)
2) mark option "Pass Auth to Upstream".
Especially item 2 was holding me from getting the tokens authenticated from the clients... Maybe @szaimen can update the guide to also add this info.

**Comment by lukasig** [comment]
That fix came fast :)   Thanks

In the mean time I played around with caddy2 and I got it to work that way quite easily. I think I'm gonna stick to that for now because the nginx proxy manager does not handle subfolders well.

**Reply by EMP83** [reply]
Hello,
I have same problem but I didn't manage to solve it. 
Could I have some help please?

**Comment by puccaso** [comment]
Hello all, 
this fix seems to only work if the domain is a private domain, as the cert needs a certain type of domain.
I am unable to get this working, using the Nginx proxy manager guide on the link above on a private domain/local/home/lan etc.. 

Any way of working around this?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one/discussions/1155#discussioncomment-3660182

**Reply by EMP83** [reply]
> Hello all, this fix seems to only work if the domain is a private domain, as the cert needs a certain type of domain. I am unable to get this working, using the Nginx proxy manager guide on the link above on a private domain/local/home/lan etc..
> 
> Any way of working around this?

for me is working like this:

added this to Nginx Proxy Manager docker-compose.yaml :

```
networks:
  default:
    external:
      name: nextcloud-aio
```

and in Nginx Proxy Manager interface i have configured like this:
![Captură de ecran 2022-10-16 155014](https://user-images.githubusercontent.com/36104474/196039126-69549434-b3d7-4d0b-919b-9ef1ab729bc4.jpg)

**Reply by puccaso** [reply]
hello @szaimen , checking that - and thank you.
@EMP83  - are you adding this to the _nextcloud_ docker-compose file or the _nginx proxy manager_ docker-compose? 
kindly, 

puc

**Reply by EMP83** [reply]
Network config i have added to the Nginx Proxy manager docker config file

**Reply by puccaso** [reply]
ah Ok. I can see a .com as your domain though, which works. I wonder if that would succeed with a fake .home domain.
